hq/app/schedule/IamJob.scala

```diff
@@ -28,11 +29,17 @@ class IamJob(enabled: Boolean, cacheService: CacheService, snsClient: AmazonSNSA
     val credsReport: Map[AwsAccount, Either[FailedAttempt, CredentialReportDisplay]] = getCredsReport(cacheService)
     logger.info(s"successfully collected credentials report for $id. Report is empty: ${credsReport.isEmpty}.")
 
-    makeIamNotification(getFlaggedCredentialsReports(credsReport, dynamo)).foreach { notification: IamNotification =>
+    def sendNotification(notification: Notification, users: Seq[IamAuditUser]): Unit = {
       for {
         _ <- send(notification, topicArn, snsClient, testMode)
-        _ = dynamo.putAlert(notification.iamUser)
+        _ = users.map(dynamo.putAlert)
       } yield ()
     }
+
+    makeNotification(getFlaggedCredentialsReports(credsReport, dynamo)).foreach { iamNotification =>
+      //TODO fix option handling
+      if (iamNotification.warningNotification.nonEmpty) sendNotification(iamNotification.warningNotification.get, iamNotification.alertedUsers)
```

Comment:
`iamNotification.warningNotification.map(wn => sendNotification(wn, iamNotification.alertedUsers))` and I think you can do the same thing below

Reply:
Thanks for this 👍🏽. 
IntelliJ is being silly and complaining that map shouldn't return Unit. I read about using pattern matching instead: https://stackoverflow.com/questions/13268017/whats-the-idiomatic-way-to-do-a-map-getorelse-returning-unit-in-scala. If you feel really against this, let me know. Otherwise, this is all working and tested on PROD 🎉